prediction_market_agent/agents/microchain_agent/blockchain/contract_class_converter.py

```diff
@@ -0,0 +1,143 @@
+import json
+from typing import Any
+
+import requests_cache
+from eth_typing import ChecksumAddress
+from microchain import Function
+from prediction_market_agent_tooling.tools.contract import ContractOnGnosisChain
+from web3 import Web3
+
+from prediction_market_agent.agents.microchain_agent.blockchain.code_interpreter import (
+    CodeInterpreter,
+)
+from prediction_market_agent.agents.microchain_agent.blockchain.models import (
+    AbiItemTypeEnum,
+    ABIMetadata,
+)
+
+TYPE_MAPPING: dict[str, (str, Any)] = {
```

Comment:
This is mapping from solidity type to (python type, example), right? I'm not that familiar with solidity, so need some explaining, is it tractable to manually list all the types here? You can have `struct`s in solidity, right? So you'll need to dynamically create a python data class, and add it to the mapping, depending on the contract?

Also, how do `mapping`s (the solidity construct) come into this? e.g. in the wxdai contract:
```
    mapping (address => uint)                       public  balanceOf;
    mapping (address => mapping (address => uint))  public  allowance;
```

Reply:
1. If structs are passed, I plan to fail gracefully - the solution would be to create those classes also from the ABI ([similar to what Typechain does](https://github.com/dethcrypto/TypeChain/issues/398))
2. mappings themselves can be called as methods (imagine the `get` from dict), this will be implemented in this PR